src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,164 @@
+package duke;
+
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
+/**
+ * Stores the list of tasks.
+ * Saves/loads tasks from a data file on shutdown/startup.
+ * Handles data manipulation
+ */
+public class Storage {
+
+    private static ArrayList<Task> taskArray = new ArrayList<>();
+
+    /**
+     * Loads tasks from data file if it exists
+     */
+    public Storage() {
+        try {
+            // Get directory of data
+            Path path = Paths.get("./data");
+
+            // Make new directory if it doesn't exist
+            if (!Files.exists(path)) {
+                Files.createDirectories(path);
+                System.out.println("Directory is created!");
+            }
+
+        } catch (IOException e) {
+            System.err.println("Failed to create directory!" + e.getMessage());
+        }
+
+        try {
+            FileReader fr = new FileReader("./data/duke.txt");
+            int c;
+            String savedTasks = "";
+            while ((c=fr.read()) != -1) {
+                savedTasks += (char) c;
+            }
```

Comment:
This part seems a lot like magic, including a cast of an `int` to a `char` as well as an assignment operator's return value being used.

Do you think that this part could be made more maintainable?